internal/datastore/memdb/caveat.go

```diff
@@ -66,7 +66,8 @@ func (r *memdbReader) readUnwrappedCaveatByName(tx *memdb.Txn, name string) (*co
 
 func (r *memdbReader) ListCaveats(_ context.Context, caveatNames ...string) ([]*core.CaveatDefinition, error) {
 	if !r.enableCaveats {
-		return nil, fmt.Errorf("caveats are not enabled")
```

Comment:
didn't we add this for the experimental flag? clients wouldn't be getting an error if they disabled caveats. `ListCaveats` is used in `WriteSchema`. Could you add a test making sure it does not break anything?

Reply:
The tests break without this; we are now calling ListCaveats in WriteSchema, so if caveats were disabled, then this would return an error. Instead, I just now ignore it entirely so that the WriteSchema will function properly.

Reply:
hrm, so what's the plan with the flag then? Either we honor it and make sure we truly disable caveat behaviour, or we remove the flag?

Reply:
Just prevent writing of caveats; no writing, no using